src/app/quizPage.js

```diff
@@ -0,0 +1,222 @@
+import {
+    QuestionService
+} from "../service/QuestionService.js"
+
+import {
+    QUIZ_PAGE_STYLES,
+    START_PAGE_STYLES,
+    TIMEOUT_AFTER_ANSWER_SELECTION
+} from "./appSettings.js"
+
+// will be filledi with mode object during page rendering
+let CURRENT_MODE = null; 
+let GENERATOR = null;
+
+// use to render the page for the first time, after the game start
+export function renderQuizPage(mode) {
+    const appScreen = document.querySelector('#pokequiz-app');
+    appScreen.classList.add(QUIZ_PAGE_STYLES.quizPageClass)
+    appScreen.classList.remove(START_PAGE_STYLES.startPageClass)
+    const quizTemplate = document.getElementById('quiz-template');
+    appScreen.innerHTML = quizTemplate.innerHTML;
+    // TODO later  - generate question using questionService - below are temporary dummy variables
+    const generatedQuestion = {
+        question: "quizQuestion",
+        questionNum: 1,
+    }
+    setupPageTitle(mode);
+    CURRENT_MODE = mode;
+    //GENERATOR = new QuestionService.Generator()
+    //TODO setupTimer() -- here or directly in App
+    renderNextQuestion(CURRENT_MODE);
+}
+
+
+// use to update quizPage and change only the question, new answers and question counter
+// not changing the timer and bar
+// gent question generator and use generates next question if there is any left to answer
+// otherwise finishes the game and redirect user to the summary page
+export function renderNextQuestion(mode) {
+    //genQuestion nie powinno być przekazywane do funkcji tylko powinno być tu wywoływana
+    const genQuestion = getNextQuestion(mode); // TODO later pass generator and use  generator.genQuestion(), and replace dummy function with real one, once it's implemented
+    if (genQuestion) { // some questions still left to answer
+        const quizBody = document.querySelector("#quiz-body");
+        // Update question
+        const quizQuestionElem = quizBody.querySelector(".quiz-question");
+        updateQuestion(quizQuestionElem, genQuestion.question, mode);
+        // Update answers list
+        const quizUl = quizBody.querySelector(".quiz-answers-list");
+        updateAnswersList(quizUl, genQuestion.question, mode);
+        // Update question counter
+        const questionCounter = document.querySelector("#question-counter");
+        updateQuestionCounter(questionCounter, genQuestion.questionNum);
+        // listen for an answer selection
+        const answersOptions = [...quizBody.querySelector(".quiz-answers-list").children]
+        for (let option of answersOptions) {
+            option.addEventListener("mouseup", function selectEventFunc() {
```

Comment:
mouseup? czemu nie click?

Reply:
a, dobra, bo może ktoś by się w ostatniej chwili cofnął i jednak chciał ściągnąć myszkę z odpowiedzi. Make sense 😌

Reply:
Exactly, that was the idea behind